src/Compilers/CSharp/Portable/Symbols/Metadata/PE/PETypeParameterSymbol.cs

```diff
@@ -301,6 +301,40 @@ public override bool HasReferenceTypeConstraint
             }
         }
 
+        internal override bool? IsNotNullableIfReferenceType
+        {
+            get
+            {
+                if ((_flags & (GenericParameterAttributes.NotNullableValueTypeConstraint | GenericParameterAttributes.ReferenceTypeConstraint)) == 0)
+                {
+                    PEModule module = ((PEModuleSymbol)this.ContainingModule).Module;
+                    GenericParameterConstraintHandleCollection constraints;
+
+                    try
+                    {
+                        constraints = module.MetadataReader.GetGenericParameter(_handle).GetConstraints();
+                    }
+                    catch (BadImageFormatException)
+                    {
+                        constraints = default(GenericParameterConstraintHandleCollection);
+                        Interlocked.CompareExchange(ref _lazyConstraintsUseSiteErrorInfo, new CSDiagnosticInfo(ErrorCode.ERR_BindToBogus, this), CSDiagnosticInfo.EmptyErrorInfo);
+                    }
```

Comment:
Please extract this to a helper method, shared with `GetDeclaredConstraintTypes()`.